### PR TITLE
Set debug log level based on a flag / env var

### PR DESCRIPTION
**Pull request by mkretz**
If we do this in the Before routine of the app we don't need to pass down the debug flag to the actions.

### CODE REVIEW

**Comment by bygui86** [comment]
I'm not able to submit my comments...

File: main.go [hunk 1]
```diff
@@ -11,15 +11,19 @@ import (
 
 const (
 	version  = "0.0.1"
-	logLevel = "debug"
+	logLevel = "info"
 )
 
+var debug = false
```
Comment (bygui86) [inline main.go]: the usage of a global variable can be avoided, see the comment below on the method "addBefore"

File: main.go [hunk 2]
```diff
@@ -44,6 +48,27 @@ func createApp() *cli.App {
 	return app
 }
```
Comment (bygui86) [inline main.go]: missing comment above the method to document it

File: main.go [hunk 3]
```diff
@@ -44,6 +48,27 @@ func createApp() *cli.App {
...
 
+func addBefore(app *cli.App, debug *bool) {
```
Comment (bygui86) [inline main.go]: remove `debug *bool` as not needed anymore, see comment below

File: main.go [hunk 4]
```diff
@@ -44,6 +48,27 @@ func createApp() *cli.App {
...
+	app.Before = func(c *cli.Context) error {
+		if *debug {
```
Comment (bygui86) [inline main.go]: `*debug` can be replaced by `c.Bool("debug")`
this way we avoid the usage of a global variable, instead we take out the value of the flag from the context

File: main.go [hunk 5]
```diff
@@ -44,6 +48,27 @@ func createApp() *cli.App {
...
+}
+
+// addFlags - Add global flags to CLI application
+func addFlags(app *cli.App) {
+	app.Flags = []cli.Flag{
+		cli.BoolFlag{
+			Name:        "debug, d",
+			Usage:       "switch on debug log output",
+			EnvVar:      "KORP_LOG_DEBUG",
+			Destination: &debug,
```
Comment (bygui86) [inline main.go]: without the global variable, this line can be removed

File: main.go (hunk 4 above)
Comment (mkretz) [reply main.go]: I agree. We could use the same approach in the actions too. Will correct it for debug on this branch.

File: main.go (hunk 5 above)
Comment (mkretz) [reply main.go]: I agree.

File: main.go (hunk 2 above)
Comment (mkretz) [reply main.go]: Agree.

File: main.go (hunk 3 above)
Comment (mkretz) [reply main.go]: Jep.

**Comment by mkretz** [comment]
Will make the changes and submit a new pull request.

**Pull request closed without merging**